phpBB/includes/functions_admin.php

```diff
@@ -2573,6 +2573,11 @@ function prune($forum_id, $prune_mode, $prune_date, $prune_flags = 0, $auto_sync
 		$sql_and .= ' AND topic_status = ' . ITEM_MOVED . " AND topic_last_post_time < $prune_date";
 	}
 
+	if ($prune_limit > 0)
+	{
+		$sql_and .= " LIMIT $prune_limit";
```

Comment:
Instead of manually adding the `LIMIT`, we do actually have a separate call `$db->sql_query_limit()` that would be better used here.
You could also additionally add `$prune_limit` to the below event if you want to. That would also require a `@changed 3.1.10-RC1 Added prune_limit to event`